trie/sync.go

```diff
@@ -364,6 +371,14 @@ func trieNode(
 	return decodedNode, nil
 }
 
+func writeLeafNodeToChan(element node, ch chan core.KeyValueHolder) {
+	leafNodeElement, isLeaf := element.(*leafNode)
```

Comment:
slight optimization in case of data tries & validator tries:
```GO
func writeLeafNodeToChan(element node, ch chan core.KeyValueHolder) {
    if ch == nil{
        return
    }
    leafNodeElement, isLeaf := element.(*leafNode)
    if !isLeaf{
        return
    }
    
    trieLeaf := keyValStorage.NewKeyValStorage(leafNodeElement.Key, leafNodeElement.Value)
    ch <- trieLeaf
}
```
reason: you avoid a probable unnecessary cast

Reply:
done, thanks